pandas/_libs/groupby_helper.pxi.in

```diff
@@ -19,6 +19,18 @@ ctypedef fused rank_t:
     object
 
 
+cdef inline bint _treat_as_na(rank_t val, bint is_datetimelike) nogil:
```

Comment:
Is this only applicable in groupby or should it go in util?

Reply:
Only here.  As I mention in comments below, I'm not wild about the fact that is_datetimelike is effectively hard-coded to True in several places.  Will try to see if that causes problems in follow-up(s)